### ISSUE TITLE
Update component to remove UIActivityIndicatorView

### ISSUE THREAD

**Issue by optimviewSkycity** [facebook/componentkit#498]
I have a component which loads a picture using SDWebImage. In the +(instancetype) newWith ... class method, I wrote a custom [CKComponentViewAttribute](https://github.com/facebook/componentkit/blob/master/ComponentKit/Core/CKComponentViewAttribute.h) which sets the image on the imageView created by a trivial CKComponent which has the [UIImageView class]. At the same time, here, I check the component's state and if it is "NO" then I initialize a CKComponent with an advanced view of UIActivityIndicatorView and I center it on both the X and Y axis. In the SDWebImage's completion block, I want to update the component using the information provided [here](http://componentkit.org/docs/state.html) in order to remove the UIActivityIndicatorView, but I do not have a reference because I didn't call [super newWithComponent] yet.
Basically, I want to remove the UIActivityIndicatorView after the image is downloaded using SDWebImage. How can I accomplish that?

**Comment by ocrickard** [comment]
We provided an implementation of a super simple network image component in the open source repo: https://github.com/facebook/componentkit/blob/master/ComponentKit/Components/CKNetworkImageComponent.mm

You have a couple options:

1. Leave the activity indicator logic in the view itself, similar to how we do downloading and view updating in the network image component via a wrapper object which encapsulates both the downloader and information needed for the download. You may need to write a couple of wrapper objects to make this play nicely with SDWebImage.
2. Create a component that encapsulates loading state, and when you receive the image updateState to go from a loading state to a non-loading state, and create a child image component instead of a child activity indicator component.

Option 1 is a better bet in my opinion, and has the added benefit of being more performant for complex component hierarchies, though I think Option 2 may be a bit more pure.
